<launch>
    <arg name="imu_name" default="imu" />
    <arg name="device" default="/dev/imu_riptide" />
    <arg name="settings" value="$(find riptide_hardware)/cfg/imu_columbus.yaml"/>
    <arg name="post_IIR_LPF_bandwidth" default="17" />

    <group ns="$(arg imu_name)">
        <rosparam command="load" file="$(arg settings)" />
        <param name="imu_name" type="string" value="$(arg imu_name)" />
        <param name="frame_id" type="string" value="$(arg imu_name)"/>
        <param name="device" type="string" value="$(arg device)" />
        
        <node pkg="imu_3dm_gx4" name="$(arg imu_name)" type="imu_3dm_gx4" output="screen"/>

        <node pkg="riptide_hardware" name="imu_processor" type="imu_processor" output="screen" >
            <rosparam command="load" file="$(arg settings)" />
            <param name="imu_name" type="string" value="$(arg imu_name)" />
            <param name="post_IIR_LPF_bandwidth" type="double" value="$(arg post_IIR_LPF_bandwidth)" />
        </node>
    </group>
</launch>